database/sqlite/sqlite_metadata.c

```diff
@@ -0,0 +1,1485 @@
+                        metadata_flag_clear(wc, METADATA_FLAG_CLEANUP);
+                    }
+                    break;
+                case METADATA_UNITTEST:
+                    ;
+                    struct thread_unittest {
+                        int join;
+                        unsigned added;
+                        unsigned processed;
+                        unsigned *done;
+                    };
+
+                    struct thread_unittest *tu = (struct thread_unittest *) cmd.param[0];
+                    usleep(1000); // processing takes 1ms
+                    __atomic_fetch_add(&tu->processed, 1, __ATOMIC_SEQ_CST);
+                    break;
+                default:
+                    break;
+            }
+            if (in_transaction && (commands_in_transaction >= METADATA_MAX_TRANSACTION_BATCH || opcode != next_opcode)) {
+                in_transaction = 0;
+                db_execute("COMMIT TRANSACTION;");
+                commands_in_transaction = 0;
+            }
+
+            if (cmd.completion)
+                completion_mark_complete(cmd.completion);
+        } while (opcode != METADATA_DATABASE_NOOP);
+    }
+
+    if (!uv_timer_stop(&wc->timer_req))
+        uv_close((uv_handle_t *)&wc->timer_req, NULL);
+
+    /*
+     * uv_async_send after uv_close does not seem to crash in linux at the moment,
+     * it is however undocumented behaviour we need to be aware if this becomes
+     * an issue in the future.
+     */
+    uv_close((uv_handle_t *)&wc->async, NULL);
+    uv_run(loop, UV_RUN_DEFAULT);
+
+    /* TODO: don't let the API block by waiting to enqueue commands */
+    uv_cond_destroy(&wc->cmd_cond);
+    /*  uv_mutex_destroy(&wc->cmd_mutex); */
+    //fatal_assert(0 == uv_loop_close(loop));
+    int rc;
+
+    do {
+        rc = uv_loop_close(loop);
+    } while (rc != UV_EBUSY);
+
+    freez(loop);
+    worker_unregister();
+
+    buffer_free(work_buffer);
+    info("METADATA: Shutting down event loop");
+    completion_mark_complete(&wc->init_complete);
+    return;
+
+error_after_timer_init:
+    uv_close((uv_handle_t *)&wc->async, NULL);
+error_after_async_init:
+    fatal_assert(0 == uv_loop_close(loop));
+error_after_loop_init:
+    freez(loop);
+    worker_unregister();
+}
+
+struct metadata_wc metasync_worker;
+
+void metadata_sync_shutdown(void)
+{
+    completion_init(&metasync_worker.init_complete);
+
+    struct metadata_cmd cmd;
+    memset(&cmd, 0, sizeof(cmd));
+    info("METADATA: Sending a shutdown command");
+    cmd.opcode = METADATA_SYNC_SHUTDOWN;
+    metadata_enq_cmd(&metasync_worker, &cmd);
+
+    /* wait for metadata thread to shut down */
+    info("METADATA: Waiting for shutdown ACK");
+    completion_wait_for(&metasync_worker.init_complete);
+    completion_destroy(&metasync_worker.init_complete);
+    info("METADATA: Shutdown complete");
+}
+
+void metadata_sync_shutdown_prepare(void)
+{
+    struct metadata_cmd cmd;
+    memset(&cmd, 0, sizeof(cmd));
```

Comment:
It is not the first time I am seeing you using a `memset` to reset the variable, probably it is better to say for compiler to reset them.